tests/queries/test_qs_combinators.py

```diff
@@ -206,6 +206,24 @@ def test_union_with_values_list_and_order(self):
             with self.subTest(qs=qs):
                 self.assertEqual(list(qs), expected_result)
 
+    def test_annotated_union(self):
+        ReservedName.objects.bulk_create([
+            ReservedName(name='rn1', order=7),
+            ReservedName(name='rn2', order=5),
+            ReservedName(name='rn0', order=6),
+            ReservedName(name='rn9', order=-1),
+        ])
+        qs1 = ReservedName.objects.annotate(
+            annotated_field=Value(2, output_field=IntegerField())
+        ).filter(order__gte=6)
+        qs2 = ReservedName.objects.annotate(
+            annotated_field=Value(1, output_field=IntegerField())
```

Comment:
`Value` wrapping and `output_field` is unnecessary since on the main branch

```suggestion
            annotated_field=2,
        ).filter(order__gte=6)
        qs2 = ReservedName.objects.annotate(
            annotated_field=1,
```

or

```suggestion
            annotated_field=Value(2),
        ).filter(order__gte=6)
        qs2 = ReservedName.objects.annotate(
            annotated_field=Value(1),
```

both work